app/utils/queryBuilder.ts

```diff
@@ -420,20 +300,21 @@ const buildQuery = (
  * URL query param
  * @returns {string} a graphql querry
  */
-export const useQueryBuilder = (
+export const useQueryBuilder = <T extends Record<string, unknown>>(
```

Comment:
and here we now need to declare carry along the generic `T` so that we can accept the column arrays as  `ColDataCardDef<T>`.